homeassistant/components/cookidoo/__init__.py

```diff
@@ -49,3 +35,39 @@ async def async_setup_entry(hass: HomeAssistant, entry: CookidooConfigEntry) ->
 async def async_unload_entry(hass: HomeAssistant, entry: CookidooConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(
+    hass: HomeAssistant, config_entry: CookidooConfigEntry
+) -> bool:
+    """Migrate config entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if config_entry.version == 1 and config_entry.minor_version == 1:
+        # Add the unique uuid
+        cookidoo = await cookidoo_from_config_entry(hass, config_entry)
+
+        try:
+            auth_data = await cookidoo.login()
+        except (CookidooRequestException, CookidooAuthException) as e:
+            _LOGGER.error(
+                "Could not migrate config config_entry: %s",
+                str(e),
+            )
+            return False
+
+        hass.config_entries.async_update_entry(
```

Comment:
I wouldn't do a migration just to set a unique ID. Instead, just during setup, if unique_id is not set, set it.

Reply:
In general I think we do like to use a minor version migration for this as it centralizes the migration logic into a single place.